Modules/Microsoft365DSC/DSCResources/MSFT_AADGroupOwnerConsentSettings/MSFT_AADGroupOwnerConsentSettings.psm1

```diff
@@ -0,0 +1,552 @@
+
+        [Parameter()]
+        [System.String]
+        $CertificateThumbprint,
+
+        [Parameter()]
+        [Switch]
+        $ManagedIdentity
+    )
+
+    $ConnectionMode = New-M365DSCConnection -Workload 'MicrosoftGraph' `
+        -InboundParameters $PSBoundParameters
+
+    #Ensure the proper dependencies are installed in the current environment.
+    Confirm-M365DSCDependencies
+
+    #region Telemetry
+    $ResourceName = $MyInvocation.MyCommand.ModuleName.Replace('MSFT_', '')
+    $CommandName = $MyInvocation.MyCommand
+    $data = Format-M365DSCTelemetryParameters -ResourceName $ResourceName `
+        -CommandName $CommandName `
+        -Parameters $PSBoundParameters
+    Add-M365DSCTelemetryEvent -Data $data
+    #endregion
+
+
+    try
+    {
+        $currentDSCBlock = $null
+
+        $consentPolicySettingsTemplateId = Get-MSConsentPolicySettingsTemplateId
+        write-verbose "Get GroupSettings template with TemplateId $consentPolicySettingsTemplateId"
+        $groupConsentSettingsTemplate = Get-MgGroupSettingTemplateGroupSettingTemplate -GroupSettingTemplateId $consentPolicySettingsTemplateId
+
+        $params = @{
+            IsSingleInstance      = 'Yes'
+            Credential            = $Credential
+            ApplicationId         = $ApplicationId
+            TenantId              = $TenantId
+            ApplicationSecret     = $ApplicationSecret
+            CertificateThumbprint = $CertificateThumbprint
+            ManagedIdentity       = $ManagedIdentity
+        }
+        $results = Get-TargetResource @params
+
+        if ($results -is [System.Collections.Hashtable] -and $results.Count -gt 1)
+        {
+            Write-Host "`r`n" -NoNewline
+            Write-Host "    |---[1/1] $($groupConsentSettingsTemplate.DisplayName)" -NoNewline
+            $results = Update-M365DSCExportAuthenticationResults -ConnectionMode $ConnectionMode `
+                -Results $results
+            $currentDSCBlock = Get-M365DSCExportContentForResource -ResourceName $ResourceName `
+                -ConnectionMode $ConnectionMode `
+                -ModulePath $PSScriptRoot `
+                -Results $results `
+                -Credential $Credential
+            Save-M365DSCPartialExport -Content $currentDSCBlock `
+                -FileName $Global:PartialExportFileName
+
+            Write-Host $Global:M365DSCEmojiGreenCheckMark
+        }
+        else
+        {
+            Write-Host $Global:M365DSCEmojiRedX
+        }
+
+        return $currentDSCBlock
+
+    }
+    catch
+    {
+        Write-Host $Global:M365DSCEmojiRedX
+
+        New-M365DSCLogEntry -Message 'Error during Export:' `
+            -Exception $_ `
+            -Source $($MyInvocation.MyCommand.Source) `
+            -TenantId $TenantId `
+            -Credential $Credential
+
+        return ''
+    }
+}
+
+function Get-MSConsentPolicySettingsTemplateId
+{
+param()
+
+    # fixed GUID
+    return 'dffd5d46-495d-40a9-8e21-954ff55e198a'
```

Comment:
Why have a function just to return a GUID? Can't we just have the GUID returned directly instead to simplify?

Reply:
The GUID is used in 3 different places but using the the function it is only defined once